### PR TITLE
Dataset and repository service changes

### PR DESCRIPTION
**Pull request by anandjakhaniya**

### CODE REVIEW

File: backend/config.default.yaml [hunk 1]
```diff
@@ -99,7 +99,6 @@ migration:
 
 feature-flag:
   disabled-authz: true
-  public_sharing_enabled: false
```
Comment (anandjakhaniya) [inline backend/config.default.yaml:102]: Removed, now not in use.  it was used https://github.com/VertaAI/modeldb/pull/1773/files#diff-40269480ca4116bb3990b71cde32415e1ae1ca3c37c7de6c1d5fbf8e38738f97L168-R158 only.

**Comment by anandjakhaniya** [comment]
I know this is a still big PR but we have used the Repository structure for the Dataset so I can not divide it into multiple PRs. Sorry about that @conradoverta .

**Comment by conradoverta** [comment]
> I know this is a still big PR but we have used the Repository structure for the Dataset so I can not divide it into multiple PRs. Sorry about that @conradoverta .

It's fine. The smallest we can make them is good enough, though we could have split the public stuff.

File: backend/src/main/java/ai/verta/modeldb/dataset/DatasetDAORdbImpl.java [hunk 2]
```diff
@@ -143,16 +142,29 @@ public Dataset createDataset(Dataset dataset, UserInfo userInfo)
...
       transaction.commit();
       LOGGER.debug("Dataset created successfully");
+
+      roleService.createWorkspacePermissions(
+          dataset.getWorkspaceServiceId(),
+          Optional.of(dataset.getWorkspaceType()),
+          dataset.getId(),
+          Optional.empty(), // UAC will populate the owner ID
+          ModelDBServiceResourceTypes.DATASET,
+          CollaboratorTypeEnum.CollaboratorType.READ_ONLY,
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/dataset/DatasetDAORdbImpl.java]: This is part of the proto call, I believe. So we should fetch the custom collaborator type from there.

File: backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java [hunk 3]
```diff
@@ -462,7 +465,20 @@ public RepositoryEntity setRepository(
...
-        createRoleBindingsForRepository(repository, userInfo, repositoryEntity);
+        roleService.createWorkspacePermissions(
+            repository.getWorkspaceServiceId(),
+            Optional.of(repository.getWorkspaceType()),
+            String.valueOf(repository.getId()),
+            Optional.empty(), // UAC will populate the owner ID
+            repositoryEntity.isDataset()
+                ? ModelDBServiceResourceTypes.DATASET
+                : ModelDBServiceResourceTypes.REPOSITORY,
+            CollaboratorTypeEnum.CollaboratorType.READ_ONLY,
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java]: Same

**Review by conradoverta — CHANGES_REQUESTED** [review]

**Review by evgeniilezhnin — APPROVED** [review]

**Review by evgeniilezhnin — CHANGES_REQUESTED** [review]
it has compilation error

File: backend/src/main/java/ai/verta/modeldb/dataset/DatasetDAORdbImpl.java (hunk 2 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/dataset/DatasetDAORdbImpl.java]: Updated from custom collaborator type proto

File: backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java (hunk 3 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java]: Updated from custom collaborator type proto

**Review by conradoverta — APPROVED** [review]
Approved assuming we fix the build to pass.

**Review by coryverta — CHANGES_REQUESTED** [review]
@ad-47 the build is failing due to compilation errors

**Review by coryverta — APPROVED** [review]

**Pull request merged**